ciao-controller/compute.go

```diff
@@ -906,7 +911,16 @@ func createServer(w http.ResponseWriter, r *http.Request, context *controller) {
 	}
 	servers.TotalServers = len(instances)
 
-	b, err := json.Marshal(servers)
+	// set machine ID for OpenStack compatibility
+	server.Server.ID = instances[0].ID
```

Comment:
If instances is an empty array 907 will not run.  The entire for loop will be skipped and then we'll panic.  Now it could be that context.startWorkload will never return an empty array in which case we're okay.  But if it can return an empty array and a nil error, we'd panic here.